eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -1629,43 +1596,44 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder, val
       // if channel is private, we send the channel_update directly to remote
       // they need it "to learn the other end's forwarding parameters" (BOLT 7)
       (state, nextState, stateData, nextStateData) match {
-        case (_, _, d1: DATA_NORMAL, d2: DATA_NORMAL) if !d1.commitments.announceChannel && !d1.buried && d2.buried =>
+        case (NORMAL | SYNCING, NORMAL, d1: DATA_NORMAL, d2: DATA_NORMAL) if !d1.commitments.announceChannel && !d1.buried && d2.buried =>
           // for a private channel, when the tx was just buried we need to send the channel_update to our peer (even if it didn't change)
           send(d2.channelUpdate)
         case (SYNCING, NORMAL, d1: DATA_NORMAL, d2: DATA_NORMAL) if !d1.commitments.announceChannel && d2.buried =>
           // otherwise if we're coming back online, we rebroadcast the latest channel_update
           // this makes sure that if the channel_update was missed, we have a chance to re-send it
           send(d2.channelUpdate)
-        case (_, _, d1: DATA_NORMAL, d2: DATA_NORMAL) if !d1.commitments.announceChannel && d1.channelUpdate != d2.channelUpdate && d2.buried =>
+        case (NORMAL | SYNCING, NORMAL, d1: DATA_NORMAL, d2: DATA_NORMAL) if !d1.commitments.announceChannel && d1.channelUpdate != d2.channelUpdate && d2.buried =>
```

Comment:
If you go to that level of granularity here and make the `onTransition` smarter, then I _think_ we can make the [handler in `NORMAL` state](https://github.com/ACINQ/eclair/blob/rename-has-commitments/eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala#L702-L712) dumber so the state of the universe is preserved. We were currently _not_ generating a new `channelUpdate` on purpose so that the `d1.channelUpdate != d2.channelUpdate` condition would not be met in the `onTransition`.

⚠️  🐇 🕳️

Reply:
I'm not sure there is anything to change/simplify here. What do you mean by "he state of the universe is preserved."? We want to keep our protection against gossip spam, and I really want to avoid putting complex business logic in the transitions, I prefer having it in the handlers like we do today.

Reply:
> I'm not sure there is anything to change/simplify here. 

Okay, let's not waste too much time on this since I will revisit in #2224.

> What do you mean by "the state of the universe is preserved."?

Nothing, just a silly joke.